### PR TITLE
Feature/price formatting

### LINKED ISSUE #3553: Allow custom currency formats
## What is the motivation for adding / enhancing this feature?

Different countries have varying currency formats -- specifically the characters used to indicate the thousandth and decimal figures. Whereas in the U.S. the price of a product would be $123,456.768, in a European country such as Hungary it would look like 123.456,768 Ft.

## What are the acceptance criteria 

- [ ] Add an option in the config, in addition to the currency sign, to allow for formatting the numeric value of the currency.
- [ ] Preferably, use a node module that adheres to ISO 4217, so that simply by choosing the currency code it will automatically format the currency value (or use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat).

## Can you complete this feature request by yourself?

- [ ] YES
- [X] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a normal feature request. This should be available on https://test.storefrontcloud.io and then after tests this can be added to next Vue Storefront version. In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] (**Pick this option only if you're sure**) This is an important improvement request for current Release Candidate version on https://next.storefrontcloud.io and should be placed in next RC version. In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] (**Pick this option only if you're sure**) This is a critical improvement request for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version. In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

### PR DESCRIPTION
**Pull request by EndPositive**
### Related Issues
closes #3553 

### Short Description and Why It's Useful

This pr adds the module [currencyformatter.js](https://osrec.github.io/currencyFormatter.js/). You are now able to change the following attributes in config:

* `currencySign`: custom currency sign.
* `currencyDecimal`: custom char used to display the decimal separator.
* `currencyGroup`: custom char used to display the decimal separator.
* `currencyPattern`: custom numeric format string. This replaces the priceFormat attribute as you are able to switch sign and value position using this attribute. E.g. `#,##0.00!` and `!#,##0.00`. Furthermore, using this pattern, you can specify how many digits the fractional part of the price should have (`#,###!` for no digits).

When these values are set to `null`, the locale default will be taken. When one of these are set, the locale will be overwritten. They can be used individually or together.

### Which Environment This Relates To
- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/numeric.md) on how to port existing Vue Storefront sites with this new feature

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by andrzejewsky** [comment]
@pkarw I wanted to write the same comment as you did. 
@EndPositive it's a cool one, but our regex is totally enough, actually we can improve it somehow(?), but it's not such important that we need to an external library, it's increasing bundle size even though the value is not that big :)

There is `Intl.NumberFormat` you can move parameters like: `minimumFractionDigits` and, `maximumFractionDigits` to the config file and that will be even better and simpler :)

File: core/filters/price.js [hunk 1]
```diff
@@ -1,14 +1,21 @@
 import { currentStoreView } from '@vue-storefront/core/lib/multistore'
 
-const formatValue = (value, locale) => {
-  const price = Math.abs(parseFloat(value))
-  const formatter = new Intl.NumberFormat(locale, { minimumFractionDigits: 2, maximumFractionDigits: 2 })
```
Comment (andrzejewsky) [inline core/filters/price.js]: I see your changes, looks much better but... why not just changing this line? I mean... I think it's totally enough to move just these two options  `minimumFractionDigits` and `maximumFractionDigits` to the config file and you don't anything else... ?

Comment (EndPositive) [reply core/filters/price.js]: Because as said #3553, they also wanted to be able to change the separators.

**Comment by andrzejewsky** [comment]
@EndPositive you have an error during the build - I'm pretty sure it's because of config file - it's not valid json, please change null's to the default separators

**Comment by EndPositive** [comment]
@andrzejewsky Thanks for the heads up. I changed the config file.

**Comment by andrzejewsky** [comment]
@EndPositive thanks!

**Pull request merged**